tests/v2/validation/rbac/clusterandprojectroles/restrictedadmin_role_test.go

```diff
@@ -28,6 +28,7 @@ type RestrictedAdminTestSuite struct {
 }
 
 func (ra *RestrictedAdminTestSuite) TearDownSuite() {
+	provisioning.SetUpdateConfig(true)
```

Comment:
this might be a problem at a high level for cleanup design. If this has to be `true` for cleanup to work, I would suggest to instead create a cleanup func in the setupUpdateConfig function that looks like 
```
client.Session.RegisterCleanupFunc(func() error {
  updateConfig = true
}
```

but I am curious, why does this need to be true in order for cleanup to work properly? perhaps the addition of a godoc comment on the called function would explain this / give context for why.

Reply:
I didn't know about this RegisterCleanupFunc, it looks clean, I'll refactor to use it